code/modules/reagents/reagent_containers/medspray.dm

```diff
@@ -0,0 +1,91 @@
+/obj/item/reagent_containers/medspray
+	name = "medical spray"
+	desc = "A medical spray bottle, designed for precision application, with an unscrewable cap."
+	icon = 'icons/obj/chemical.dmi'
+	icon_state = "medspray"
+	item_state = "spraycan"
+	lefthand_file = 'icons/mob/inhands/equipment/hydroponics_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/equipment/hydroponics_righthand.dmi'
+	flags_1 = NOBLUDGEON_1
+	obj_flags = UNIQUE_RENAME
+	container_type = OPENCONTAINER
+	slot_flags = SLOT_BELT
+	throwforce = 0
+	w_class = WEIGHT_CLASS_SMALL
+	throw_speed = 3
+	throw_range = 7
+	amount_per_transfer_from_this = 10
+	volume = 60
+	var/can_fill_from_container = TRUE
+	var/apply_type = PATCH
+	var/apply_method = "spray"
+	var/self_delay = 30
+	var/squirt_mode = 0
+	var/squirt_amount = 5
+
+/obj/item/reagent_containers/medspray/attack_self(mob/user)
+	squirt_mode = !squirt_mode
+	if(squirt_mode)
+		amount_per_transfer_from_this = squirt_amount
+	else
+		amount_per_transfer_from_this = initial(amount_per_transfer_from_this)
+	to_chat(user, "<span class='notice'>You will now apply the medspray's contents in [squirt_mode ? "short bursts":"extended sprays"]. You'll now use [amount_per_transfer_from_this] units per use.</span>")
+
+/obj/item/reagent_containers/medspray/attack(mob/M, mob/user, def_zone)
+	if(!reagents || !reagents.total_volume)
+		to_chat(user, "<span class='warning'>[src] is empty!</span>")
+		return
+
+	if(M == user)
+		M.visible_message("<span class='notice'>[user] attempts to [apply_method] [src] on themselves.</span>")
+		if(self_delay)
+			if(!do_mob(user, M, self_delay))
+				return
+			if(!reagents || !reagents.total_volume)
+				return
+		to_chat(M, "<span class='notice'>You [apply_method] yourself with [src].</span>")
+
+	else
+		M.visible_message("<span class='danger'>[user] attempts to [apply_method] [src] on [M].</span>", \
+							"<span class='userdanger'>[user] attempts to [apply_method] [src] on [M].</span>")
```

Comment:
Add some logging for the attempt.
add_logs(user, M, "attempted to apply", src, reagents.log_list())